src/ripple/basics/strHex.h

```diff
@@ -62,22 +66,58 @@ charUnHex (char c)
 }
 /** @} */
 
-// NIKB TODO cleanup this function and reduce the need for the many overloads
-//           it has in various places.
-template<class FwdIt>
-std::string strHex (FwdIt first, int size)
+template<class InputIterator>
+std::string strHex(InputIterator begin, InputIterator end)
+{
+    std::string result{};
+    result.reserve (std::distance(begin, end));
+    boost::algorithm::hex (begin, end, std::back_inserter(result));
+    return result;
+}
+namespace
```

Comment:
We don't use anonymous namespaces in rippled. We may change this, but for now make this a `detail` namespace